From 7a4998f71b10a84da96b6950fd1e91282a19de29 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Tue, 21 Feb 2023 17:34:57 +0000
Subject: [PATCH] WebAppWayland: do not show window if its status is still
 hidden

Note: GetHiddenWindow takes into account the _no_window flag

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
Upstream-Status: Inappropriate [LuneOS specific]

 src/platform/web_app_wayland.cc | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/platform/web_app_wayland.cc b/src/platform/web_app_wayland.cc
index e36eaf2..6d23bae 100644
--- a/src/platform/web_app_wayland.cc
+++ b/src/platform/web_app_wayland.cc
@@ -295,12 +295,16 @@ void WebAppWayland::OnStageActivated() {
 
   Page()->ResumeWebPageAll();
 
-  Page()->SetVisibilityState(
-      WebPageBase::WebPageVisibilityState::kWebPageVisibilityStateVisible);
+  if(!GetHiddenWindow()) {
+      Page()->SetVisibilityState(
+          WebPageBase::WebPageVisibilityState::kWebPageVisibilityStateVisible);
+  }
 
   SetActiveInstanceId(InstanceId());
 
-  app_window_->Show();
+  if(!GetHiddenWindow()) {
+    app_window_->Show();
+  }
 
   LOG_INFO(MSGID_WEBAPP_STAGE_ACITVATED, 3, PMLOGKS("APP_ID", AppId().c_str()),
            PMLOGKS("INSTANCE_ID", InstanceId().c_str()),
